site/app/controllers/calendar/CalendarController.php

```diff
@@ -132,76 +128,130 @@ public function editMessage(): RedirectResponse {
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        if (isset($_POST['semester'])) {
+            $semester = $_POST['semester'];
         }
-
-        $calendar_item = $this->core->getCourseEntityManager()->getRepository(CalendarItem::class)
-            ->findOneBy(['id' => $id]);
-
-        if ($calendar_item === null) {
-            $this->core->addErrorMessage("An error has occured.");
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        else {
+            $this->core->addErrorMessage("Invalid semester");
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
 
-        $calendar_item->setText($text);
-        $calendar_item->setDate(new \DateTime($date));
-        try {
-            $calendar_item->setStringType($type);
+        if (isset($_POST['course'])) {
+            $InputCourse = $_POST['course'];
         }
-        catch (\InvalidArgumentException  $e) {
-            $this->core->addErrorMessage("That is not a valid calendar item type");
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        else {
+            $this->core->addErrorMessage("Invalid course");
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
-        $this->core->getCourseEntityManager()->flush();
-        return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+
+        $instructor_courses = $this->core->getQueries()->getInstructorLevelUnarchivedCourses($this->core->getUser()->getId());
+
+        foreach ($instructor_courses as $course) {
+            if (($semester === $course['semester']) && ($InputCourse === $course['course'])) {
+                $this->core->loadCourseConfig($course['semester'], $course['course']);
+                $this->core->loadCourseDatabase();
+                $calendar_item = $this->core->getCourseEntityManager()->getRepository(CalendarItem::class)
+                    ->findOneBy(['id' => $id]);
+                if ($calendar_item === null) {
+                    return new RedirectResponse($this->core->buildUrl(['calendar']));
+                }
+                try {
+                    $calendar_item->setText($text);
+                }
+                catch (\InvalidArgumentException $e) {
+                    $this->core->addErrorMessage($e->getMessage());
+                    return new RedirectResponse($this->core->buildUrl(['calendar']));
+                }
+                $calendar_item->setDate(new \DateTime($date));
+                try {
+                    $calendar_item->setStringType($type);
+                }
+                catch (\InvalidArgumentException  $e) {
+                    $this->core->addErrorMessage($e->getMessage());
+                    return new RedirectResponse($this->core->buildUrl(['calendar']));
+                }
```

Comment:
You could consolidate the two try catches by putting the statements inside both try's inside one try catch instead.